diff --git a/allowed_formats.post_update.php b/allowed_formats.post_update.php
index 86019e5..808eac3 100644
--- a/allowed_formats.post_update.php
+++ b/allowed_formats.post_update.php
@@ -15,7 +15,7 @@ function allowed_formats_post_update_store_allowed_formats_as_sequence() {
     /** @var \Drupal\field\Entity\FieldConfig $field_config */
     if (in_array($field_config->getType(), _allowed_formats_field_types(), TRUE)) {
       $allowed_formats = $field_config->getThirdPartySettings('allowed_formats');
-      if (!empty($allowed_formats)) {
+      if (!empty($allowed_formats) && !array_key_exists('allowed_formats', $allowed_formats)) {
         // Unset existing configuration.
         foreach ($allowed_formats as $key => $value) {
           $field_config->unsetThirdPartySetting('allowed_formats', $key);
