diff --git a/src/Plugin/EntityReferenceSelection/TermSelection.php b/src/Plugin/EntityReferenceSelection/TermSelection.php
index 8d62735..ec5df36 100644
--- a/src/Plugin/EntityReferenceSelection/TermSelection.php
+++ b/src/Plugin/EntityReferenceSelection/TermSelection.php
@@ -29,7 +29,7 @@ class TermSelection extends DefaultSelection {
       '#type' => 'radios',
       '#title' => $this->t('Target vocabulary'),
       '#options' => $vocab_storage->getVocabsList(),
-      '#default_value' => isset($this->configuration['handler_settings']['target_vocab']) ? $this->configuration['handler_settings']['target_vocab'] : NULL,
+      '#default_value' => isset($this->configuration['target_vocab']) ? $this->configuration['target_vocab'] : NULL,
       '#required' => TRUE,
     ];
 
@@ -49,7 +49,7 @@ class TermSelection extends DefaultSelection {
     /** @var \Drupal\config_terms\TermStorageInterface $term_storage */
     $term_storage = $this->entityTypeManager->getStorage('config_terms_term');
 
-    $handler_settings = $this->configuration['handler_settings'];
+    $handler_settings = $this->configuration;
     $vid = isset($handler_settings['target_vocab']) ? $handler_settings['target_vocab'] : FALSE;
 
     // The return array needs to be keyed by bundle. There is only one.
