diff --git a/modules/hdbt_admin_editorial/hdbt_admin_editorial.module b/modules/hdbt_admin_editorial/hdbt_admin_editorial.module
index 968464f..da0a797 100644
--- a/modules/hdbt_admin_editorial/hdbt_admin_editorial.module
+++ b/modules/hdbt_admin_editorial/hdbt_admin_editorial.module
@@ -128,14 +128,15 @@ function hdbt_admin_editorial_form_taxonomy_term_form_alter(array &$form, FormSt
  * Implements hook_field_widget_WIDGET_TYPE_form_alter().
  *
  * Hero designs:
- *   background-image     = "Background image"
- *   diagonal             = "Diagonal"
- *   with-image-bottom    = "Image on the bottom"
- *   with-image-left      = "Image on the left"
- *   with-image-right     = "Image on the right"
- *   without-image-center = "Without image, align center"
- *   without-image-left   = "Without image, align left"
- *   with-search          = "With search"
+ *   background-image               = "Background image"
+ *   background-image-with-text-box = "Background image with text box"
+ *   diagonal                       = "Diagonal"
+ *   with-image-bottom              = "Image on the bottom"
+ *   with-image-left                = "Image on the left"
+ *   with-image-right               = "Image on the right"
+ *   without-image-center           = "Without image, align center"
+ *   without-image-left             = "Without image, align left"
+ *   with-search                    = "With search"
  */
 function hdbt_admin_editorial_field_widget_paragraphs_form_alter(&$element, &$form_state, $context) {
 
@@ -156,6 +157,8 @@ function hdbt_admin_editorial_field_widget_paragraphs_form_alter(&$element, &$fo
         [$design_select => ['value' => 'background-image']],
         'or',
         [$design_select => ['value' => 'with-search']],
+        'or',
+        [$design_select => ['value' => 'background-image-with-text-box']],
       ],
     ];
 
@@ -172,6 +175,8 @@ function hdbt_admin_editorial_field_widget_paragraphs_form_alter(&$element, &$fo
     $link_states = [
       'visible' => [
         [$design_select => ['value' => 'background-image']],
+        'or',
+        [$design_select => ['value' => 'background-image-with-text-box']],
       ],
     ];
     $element['subform']['field_hero_link']['#states'] = $link_states;
