--- a/src/Normalizer/RefreshTokenEntityNormalizer.php	2026-04-17 14:55:28
+++ b/src/Normalizer/RefreshTokenEntityNormalizer.php	2026-04-17 14:55:39
@@ -11,12 +11,21 @@
    *
    * @var string|array
    */
-  protected $supportedInterfaceOrClass = '\League\OAuth2\Server\Entities\RefreshTokenEntityInterface';
+  protected $supportedInterfaceOrClass = 'League\OAuth2\Server\Entities\RefreshTokenEntityInterface';
 
   /**
    * {@inheritdoc}
    */
-  public function normalize($token_entity, $format = NULL, array $context = []) {
+  public function getSupportedTypes(?string $format): array {
+    return [
+      $this->supportedInterfaceOrClass => TRUE,
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function normalize(mixed $token_entity, ?string $format = NULL, array $context = []): array {
     /** @var \League\OAuth2\Server\Entities\TokenInterface $token_entity */
     return [
       'value' => $token_entity->getIdentifier(),
--- a/src/Normalizer/UserClaimsNormalizer.php	2026-04-17 14:55:28
+++ b/src/Normalizer/UserClaimsNormalizer.php	2026-04-17 14:55:43
@@ -20,6 +20,15 @@
   protected $format = 'json';
 
   /**
+   * {@inheritdoc}
+   */
+  public function getSupportedTypes(?string $format): array {
+    return [
+      $this->supportedInterfaceOrClass => TRUE,
+    ];
+  }
+
+  /**
    * The user storage.
    *
    * @var \Drupal\user\UserStorageInterface
@@ -62,7 +71,7 @@
   /**
    * {@inheritdoc}
    */
-  public function normalize($user_entity, $format = NULL, array $context = []) {
+  public function normalize(mixed $user_entity, ?string $format = NULL, array $context = []): array {
     assert($user_entity instanceof UserEntityWithClaims);
     $identifier = $user_entity->getIdentifier();
     // Check if the account is in $context. If not, load it from the database.
--- a/src/Normalizer/TokenEntityNormalizer.php	2026-04-17 14:55:28
+++ b/src/Normalizer/TokenEntityNormalizer.php	2026-04-17 14:55:37
@@ -12,9 +12,18 @@
    *
    * @var string|array
    */
-  protected $supportedInterfaceOrClass = '\League\OAuth2\Server\Entities\TokenInterface';
+  protected $supportedInterfaceOrClass = 'League\OAuth2\Server\Entities\TokenInterface';
 
   /**
+   * {@inheritdoc}
+   */
+  public function getSupportedTypes(?string $format): array {
+    return [
+      $this->supportedInterfaceOrClass => TRUE,
+    ];
+  }
+
+  /**
    * @var \Drupal\Core\Entity\EntityTypeManagerInterface
    */
   protected $entityTypeManager;
@@ -29,7 +38,7 @@
   /**
    * {@inheritdoc}
    */
-  public function normalize($token_entity, $format = NULL, array $context = []) {
+  public function normalize(mixed $token_entity, ?string $format = NULL, array $context = []): array {
     /** @var \League\OAuth2\Server\Entities\TokenInterface $token_entity */
 
     $scopes = array_map(function ($scope_entity) {
