diff --git a/src/Normalizer/RefreshTokenEntityNormalizer.php b/src/Normalizer/RefreshTokenEntityNormalizer.php
index 644cf6c..f7a2c8b 100644
--- a/src/Normalizer/RefreshTokenEntityNormalizer.php
+++ b/src/Normalizer/RefreshTokenEntityNormalizer.php
@@ -14,4 +14,13 @@ class RefreshTokenEntityNormalizer extends NormalizerBase implements TokenEntityNormalizerInterface {
   */
   protected $supportedInterfaceOrClass = '\League\OAuth2\Server\Entities\RefreshTokenEntityInterface';
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getSupportedTypes(?string $format): array {
+    return [
+      $this->supportedInterfaceOrClass => TRUE,
+    ];
+  }
+
   /**
    * {@inheritdoc}
    */
diff --git a/src/Normalizer/TokenEntityNormalizer.php b/src/Normalizer/TokenEntityNormalizer.php
index c3f0df7..9a2e1b1 100644
--- a/src/Normalizer/TokenEntityNormalizer.php
+++ b/src/Normalizer/TokenEntityNormalizer.php
@@ -15,4 +15,13 @@ class TokenEntityNormalizer extends NormalizerBase implements TokenEntityNormalizerInterface {
   */
   protected $supportedInterfaceOrClass = '\League\OAuth2\Server\Entities\TokenInterface';
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getSupportedTypes(?string $format): array {
+    return [
+      $this->supportedInterfaceOrClass => TRUE,
+    ];
+  }
+
   /**
    * @var \Drupal\Core\Entity\EntityTypeManagerInterface
    */
diff --git a/src/Normalizer/UserClaimsNormalizer.php b/src/Normalizer/UserClaimsNormalizer.php
index cf3b128..e5ce6db 100644
--- a/src/Normalizer/UserClaimsNormalizer.php
+++ b/src/Normalizer/UserClaimsNormalizer.php
@@ -18,4 +18,13 @@ class UserClaimsNormalizer extends NormalizerBase implements NormalizerInterface {
 
   protected $format = 'json';
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getSupportedTypes(?string $format): array {
+    return [
+      $this->supportedInterfaceOrClass => TRUE,
+    ];
+  }
+
   /**
    * The user storage.
    *
