diff --git a/src/Controller/Oauth2Token.php b/src/Controller/Oauth2Token.php
index a00f01b..9811498 100644
--- a/src/Controller/Oauth2Token.php
+++ b/src/Controller/Oauth2Token.php
@@ -69,7 +69,7 @@ class Oauth2Token extends ControllerBase {
       $response = $this->handleToken($request, $auth_server);
     }
     catch (OAuthServerException $exception) {
-      watchdog_exception('simple_oauth', $exception);
+      $this->getLogger('simple_oauth')->error($exception->getMessage(), ['exception' => $exception]);
       $response = $exception->generateHttpResponse(new Response());
     }
     return $response;
